onnxruntime/contrib_ops/cuda/bert/add_bias_transpose.cu

```diff
@@ -518,8 +622,35 @@ template <typename T>
 void InvokeAddBiasTranspose(
     cudaStream_t stream, const int num_matrices, const int format, const int max_threads_per_block,
     const int batch_size, const int sequence_length, const int num_heads, const int qk_head_size,
-    const T* input, const T* biases, T* output, T* qkv_add_bias, const int v_head_size, int total_matrix_count) {
+    const T* input, const T* biases, T* output, T* qkv_add_bias, const int v_head_size, int total_matrix_count,
+    bool do_rotary = false, int original_past_sequence_length = 0) {
   assert(num_heads <= max_threads_per_block);
+
+  if (do_rotary) {
+#ifdef USE_ROCM
+    ORT_THROW("Rotary attention is not supported on ROCm");
+#elif defined(__CUDA_ARCH__) && __CUDA_ARCH__ > 520
+    if (format != 1 && format != 3) {
```

Comment:
Why limit it to format 1 and format 3?

Reply:
it corresponds to gpt2 case with unfused kernel format and cutlass kernel format

Reply:
we can still enable fused kernel for the context run.

Reply:
yeah we can extend the support to all formats later